osquery/tables/system/freebsd/xattr_utils.cpp

```diff
@@ -0,0 +1,176 @@
+  if (read_bytes == -1) {
+    return false;
+  }
+
+  return true;
+}
+
+bool getExtendedAttributeList(std::vector<std::string>& attribute_list,
+                              const std::string& path,
+                              bool user_namespace) {
+  attribute_list.clear();
+
+  std::string raw_attribute_list;
+  if (!getRawExtendedAttributeList(raw_attribute_list, path, user_namespace)) {
+    return false;
+  }
+
+  size_t index = 0U;
+  while (index < raw_attribute_list.size()) {
+    auto name_length = static_cast<size_t>(raw_attribute_list[index]);
+    index++;
+
+    if (index + name_length > raw_attribute_list.size()) {
+      return false;
+    }
+
+    auto current_attribute_name = raw_attribute_list.substr(index, name_length);
+    index += name_length;
+
+    attribute_list.push_back(current_attribute_name);
+  }
+
+  return true;
+}
+
+bool getExtendedAttribute(std::string& value,
+                          const std::string& path,
+                          bool user_namespace,
+                          const std::string& name) {
+  value.clear();
+
+  int namespace_ =
+      user_namespace ? EXTATTR_NAMESPACE_USER : EXTATTR_NAMESPACE_SYSTEM;
+
+  auto buffer_size =
+      extattr_get_file(path.data(), namespace_, name.data(), nullptr, 0U);
+  if (buffer_size == -1) {
+    return false;
+  }
+
+  value.resize(static_cast<size_t>(buffer_size));
+  if (value.size() != static_cast<size_t>(buffer_size)) {
+    return false;
+  }
+
+  buffer_size = extattr_get_file(
+      path.data(), namespace_, name.data(), &value[0], value.size());
+  if (buffer_size == -1) {
+    return false;
+  }
+
+  if (value.size() != static_cast<size_t>(buffer_size)) {
+    value.resize(buffer_size);
+  }
+
+  return true;
+}
+
+bool appendExtendedAttributesForNamespace(ExtendedAttributes& attributes,
+                                          const std::string& path,
+                                          bool user_namespace) {
+  std::vector<std::string> attribute_list;
+  if (!getExtendedAttributeList(attribute_list, path, user_namespace)) {
+    return false;
+  }
+
+  std::string name_prefix = user_namespace ? "user." : "system.";
+
+  for (const auto& attribute_name : attribute_list) {
+    std::string attribute_value;
+    if (!getExtendedAttribute(
+            attribute_value, path, user_namespace, attribute_name)) {
+      return false;
+    }
+
+    attributes.push_back(
+        std::make_pair(name_prefix + attribute_name, attribute_value));
+  }
+
+  return true;
+}
+} // namespace
+
+bool getExtendedAttributes(ExtendedAttributes& attributes,
+                           const std::string& path) {
+  attributes.clear();
+
+  if (!appendExtendedAttributesForNamespace(attributes, path, true)) {
+    return false;
+  }
+
+  if (!appendExtendedAttributesForNamespace(attributes, path, false)) {
+    return false;
+  }
+
+  return true;
+}
+
+bool setExtendedAttributes(
```

Comment:
This sets extended attributes right?

That would mutute system state and is an antipattern.

Reply:
Yes! This function is only used by the tests though, and is not called elsewhere

Reply:
Why are we testing it if we're not using?

Reply:
The extended_attributes_tests.cpp file sets some example extended attributes so that it can then try to read them; the same approach is already performed on master but it only works for macOS